xonsh/environ.py

```diff
@@ -1985,10 +1985,11 @@ def __getitem__(self, key):
         ):
             self._detyped = None
 
-        validator = self.get_validator(key)
-        converter = self.get_converter(key)
-        if not validator(val):
-            val = converter(val)
+        if val:
```

Comment:
You can validate against None, empty lists, empty strings etc

Reply:
Yeah, this should be the None check. Thanks! 
Pushed.